### ISSUE TITLE
Why to freeze BatchNorm after the second epoch 

### ISSUE THREAD

**Issue by Jermmy** [HuiZeng/Grid-Anchor-based-Image-Cropping-Pytorch#2]
Hi, thanks for your excellent work! 
After running the training code, I found you set `net.eval()` in `test` function but didn't set `net.train()` after evaluation, which means the BatchNorm would be freezed after the first training epoch. Is this some training trick or a bug?

**Comment by HuiZeng** [comment]
Hi，thanks for your comments.
I have checked code and found this is indeed a bug. 
Currently, I am not sure how much this bug will affect the final performance. I will check it asap.

**Comment by HuiZeng** [comment]
Hi,
It seems that BN affects much of the performance. It is better to freeze the BN either from the second epoch or from the first epoch.
The reason may be that batch size is only 1 for our task, making BN very unstable.

**Comment by Jermmy** [comment]
OK, thanks for your reply!
